ui.pyw

```diff
@@ -213,17 +216,21 @@ class DbTab(Tab):
         for item in available_restaurants:
             self.restaurants_list.insert(END, item)
 
-    def _add_person_to_db(self, event):
+    def _add_person_to_db(self):
+        d = AddUserDialog(self, self._db)
+        self.wait_window(d)
+        return
+
+    def _edit_person_in_db(self):
         user = self.choose_user_value.get().strip()
         if not user:
-            return
-        try:
-            self._db.add_person(user)
-            self.refresh()
-        except Exception as e:  #TODO catch specific exceptions and handle appropriately
-            tkMessageBox.showerror("Adding a person failed", "Either this person already exits or connecting to the database failed")
+            tkMessageBox.showinfo("User must be selected", "Please select a user from the combo box")
             return
 
+        d = EditUserDialog(self, self._db, user)
+        self.wait_window(d)
+        return
```

Comment:
an empty return at the end of the function is like an empty vacation on an empty mountain...